code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,461 @@
+/obj/item/xenoartifact
+	name = "xenoartifact"
+	icon = 'icons/obj/xenoarchaeology/xenoartifact.dmi'
+	icon_state = "map_editor"
+	w_class = WEIGHT_CLASS_NORMAL
+	light_color = LIGHT_COLOR_FIRE
+	desc = "A strange alien artifact. What could it possibly do?"
+	throw_range = 3
+	
+	var/charge = 0 //How much input the artifact is getting from activator traits
+	var/charge_req //This isn't a requirement anymore. This just affects how effective the charge is
+	var/process_type //Processing type, used for tick
+	var/list/true_target = list() //List of targted entities for traits
+
+	var/material //Associated traits & colour
+	var/list/traits = list() //activation trait, minor 1, minor 2, minor 3, major, malfunction
+	var/datum/xenoartifact_trait/touch_desc
+	var/special_desc = "The Xenoartifact is made from a" //used for special examine circumstance, science goggles
+	var/max_range = 1 //How far the artifact can reach
+
+	var/code //Used for signaler trait
+	var/frequency
+	var/datum/radio_frequency/radio_connection
+
+	var/usedwhen //holder for worldtime
+	var/cooldown = 8 SECONDS //Time between uses
+	var/cooldownmod = 0 //Extra time traits can add to the cooldown
+	COOLDOWN_DECLARE(xenoa_cooldown)
+
+	var/list/icon_slots[4] //Associated with random sprite stuff. It's setup as [4] so it's easier to check for slots being assigned by traits and such
+	var/mutable_appearance/icon_overlay 
+
+	var/malfunction_chance //Everytime the artifact is used this increases. When this is successfully proc'd the artifact gains a malfunction and this is lowered.
+	var/malfunction_mod = 1 //How much the chance can change in a sinlge itteration
+
+	var/logging = TRUE //Can be toggled by admins if it's disruptive to logs
```

Comment:
```suggestion
```
remove it
If anything this should be a config option and I believe it already is